pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -61,6 +74,17 @@ func NewNode(constraints *v1alpha5.Constraints, instanceTypeOptions []cloudprovi
 	}
 	return n
 }
+
+// TODD(todo): clean up this usage of two different set libraries
+func containsAny(get sets.Set, systems sset.String) bool {
```

Comment:
I think it will be easier to just convert sset to the karpenter set.
```
s := sets.NewSet(systems.UnsortedList()...)

Reply:
Yeah, that's cleaner. Modified.